### ISSUE TITLE
Workflow steps target_project_name mismatch

### ISSUE THREAD

**Issue by hennevogel** [openSUSE/open-build-service#11854]
The link_package step uses `:` between org and repo, the configure step uses `-`. 

```diff
home:bs-team:openSUSE:open-build-service:PR-11852
home:bs-team:openSUSE-open-build-service:PR-11852
```

Means you can't use those two steps together.

![Screenshot from 2021-11-12 18-11-41](https://user-images.githubusercontent.com/514785/141507051-dcf4fe00-34bf-426d-a70c-7b101ff95d01.png)

**Comment by hennevogel** [comment]
`Workflow::Step::ConfigureRepositories` uses it's own method for `target_project_name`. It should not.

**Comment by hennevogel** [comment]
This is stopping our own workflow runs since we merged #11846 

For an example of this problem see #11852 and https://github.com/openSUSE/open-build-service/settings/hooks/302983521/deliveries
